exp-models/addon/utils/eligibility.js

```diff
@@ -0,0 +1,113 @@
+// h/t @aaxelb
+
+import Ember from 'ember';
+import moment from 'moment';
+
+let {$} = Ember;
+
+var AND = (a, b) => {
+    return function() {
+        return a(...arguments) && b(...arguments);
+    };
+};
+var OR = (a, b) => {
+    return function() {
+        return a(...arguments) || b(...arguments);
+    };
+};
+
+/**
+ * @return {integer}: number of days represented by amount/unit pair, e.g. 3 'weeks' -> 21
+ **/
+function parseDate(amount, unit='days') {
+    var singular = unit.replace(/s$/, '');
+    return moment.duration(parseFloat(amount), `${singular}s`).asDays();
+}
+
+function compileEligibilityString(elig) {
+    var parts = elig.split(/(\(|\)|and|or)/).map((p) => p.trim()).filter((p) => p.length);
```

Comment:
If I am understanding this right: it appears you have defined a grammar or mini-DSL for specifying eligibility criteria. Can you point me to an example file containing a usage, or minimum documentation on the intended clauses? There does not appear to be any new experiment test data in https://github.com/CenterForOpenScience/experimenter/pull/38

I'd like to capture basic design notes as we merge PRs, rather than try to reconstruct from memory later. A 2-3 sentence description + example would be a valuable addition to the PR description.

(Sorry for the extra work. You can call me anything on this list in return: https://mraybould.wordpress.com/2013/01/11/shakespeare-insult-kit/ )